git_machete/client.py

```diff
@@ -1199,15 +1194,15 @@ def squash(self, current_branch: str, fork_commit: str) -> None:
         # The tree (HEAD^{tree}) argument must be passed as first,
         # otherwise the entire `commit-tree` will fail on some ancient supported
         # versions of git (at least on v1.7.10).
-        squashed_sha = self.__git.popen_git(
-            "commit-tree", "HEAD^{tree}", "-p", fork_commit, "-m", earliest_full_body,
+        squashed_sha = self.__git.get_commit_tree(
```

Comment:
As with other methods, I'd abstract away the flags wherever possible. The API of `GitContext` should not expose the details about the underlying `git`-CLI-based implementation

Reply:
I'm not sure if I get you, I tried to do something about this issue. I will upload it with next push and you can decide for yourself if it's sufficient.

Reply:
☹️

Reply:
You can rework on a subsequent PR... but generally since the rework is done here, let's implement it already in a solid manner.

Reply:
In general, the user of the API, be it `get_commit_tree` or `update_ref`, should **not** be concerned with the details like flags passed to the underlying command. Although yes, technically the original issue is resolved even with exposing a suboptimal API like the current one :trollface:

Reply:
Ok, I get you now, you would like user to just call `new_method()` or maybe `new_method(only_one_arg)` and have it done all the rest of the work. I suspect that it will lead to very weird code, because we will be trying to name with words what is actually represented by git commands `options`. What is more, whenever we need slightly different set of options we will have to create new method (which also will cause naming difficulties). 

I will do it, but I suspect that in terms of maintenance we might be looking for troubles in the future. If we would  just call `branch(*args, **kwargs)` [although `kwargs` are debatable] it would be flexible for different methods and yet good enough to manage things like `flushing cache`. There would be a problem how to differentiate between run_git and popen_git, but it is a step forward.